src/dbnode/storage/namespace/options.go

```diff
@@ -109,7 +111,9 @@ func (o *options) Equal(value Options) bool {
 		o.cleanupEnabled == value.CleanupEnabled() &&
 		o.repairEnabled == value.RepairEnabled() &&
 		o.retentionOpts.Equal(value.RetentionOptions()) &&
-		o.indexOpts.Equal(value.IndexOptions())
+		o.indexOpts.Equal(value.IndexOptions()) &&
+		(o.schemaReg == nil && value.SchemaRegistry() == nil ||
```

Comment:
super nit: Could we wrap each of the && operations in a ()? I know it binds more tightly but makes me nervous

Reply:
```golang

((o.schemaReg == nil && value.SchemaRegistry() == nil) ||
(o.schemaReg != nil && o.schemaReg.Equal(value.SchemaRegistry()))
```

Also can `schemaReg` ever be nil? Seems like you always at least set it to `emptySchemaRegistry()`. Just a precaution?

Reply:
it won't be nil ever, will simplify it.